blockchain/types/tx_internal_data_ethereum_set_code.go

```diff
@@ -0,0 +1,567 @@
+// Copyright 2024 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/kaiachain/kaia/blockchain/types/accountkey"
+	"github.com/kaiachain/kaia/common"
+	"github.com/kaiachain/kaia/common/hexutil"
+	"github.com/kaiachain/kaia/crypto"
+	"github.com/kaiachain/kaia/fork"
+	"github.com/kaiachain/kaia/kerrors"
+	"github.com/kaiachain/kaia/rlp"
+)
+
+// DelegationPrefix is used by code to denote the account is delegating to
+// another account.
+var DelegationPrefix = []byte{0xef, 0x01, 0x00}
+
+// ParseDelegation tries to parse the address from a delegation slice.
+func ParseDelegation(b []byte) (common.Address, bool) {
+	if len(b) != 23 || !bytes.HasPrefix(b, DelegationPrefix) {
+		return common.Address{}, false
+	}
+	var addr common.Address
+	copy(addr[:], b[len(DelegationPrefix):])
+	return addr, true
+}
+
+// AddressToDelegation adds the delegation prefix to the specified address.
+func AddressToDelegation(addr common.Address) []byte {
+	return append(DelegationPrefix, addr.Bytes()...)
+}
+
+// TxInternalDataEthereumSetCode represents a set code transaction.
+type TxInternalDataEthereumSetCode struct {
+	ChainID           *big.Int
+	AccountNonce      uint64
+	GasTipCap         *big.Int // a.k.a. maxPriorityFeePerGas
+	GasFeeCap         *big.Int // a.k.a. maxFeePerGas
+	GasLimit          uint64
+	Recipient         *common.Address
```

Comment:
Can you just make it `common.Address` type? That's what they did with go-ethereum https://github.com/lightclient/go-ethereum/blob/eip-7702/core/types/tx_setcode.go#L59

Reply:
Yes, I've corrected it.